reference/debugging.md

```diff
@@ -0,0 +1,229 @@
+        ```
+
+In both the Deployment pod and the individual pods, take the necessary action to address any discovered issues.
+
+## Review Ambassador Logs
+
+The Ambassador logging can provide information on anything that might be abnormal or malfunctioning. While there may be a large amount of data to sort through, look for key errors such as the Ambassador process restarting unexpectedly, or a malformed Envoy configuration.
+
+You can turn on Debug mode in the [Edge Policy Console](../../about/edge-policy-console), which generates verbose logging data that can be useful when trying to find a subtle error or bug.
+
+1. Use the following command to target an individual Ambassador Pod: `kubectl get pods`
+
+    The terminal will print something similar to the following:
+
+    ```console
+    $ kubectl get pods
+    NAME                         READY     STATUS    RESTARTS   AGE
+    ambassador-85c4cf67b-4pfj2   1/1       Running   0          3m
+    ```
+
+2. Then, run the following: `kubectl logs ambassador-<pod>`
+
+The terminal will print something similar to the following:
+
+    ```console
+    $ kubectl logs ambassador-85c4cf67b-4pfj2
+    2018-10-10 12:26:50 kubewatch 0.40.0 INFO: generating config with gencount 1 (0 changes)
+    /usr/lib/python3.6/site-packages/pkg_resources/__init__.py:1235: UserWarning: /ambassador is writable by group/others and vulnerable to attack when used with get_resource_filename. Consider a more secure location (set with .set_extraction_path or the PYTHON_EGG_CACHE environment variable).
+    warnings.warn(msg, UserWarning)
+    2018-10-10 12:26:51 kubewatch 0.40.0 INFO: Scout reports {"latest_version": "0.40.0", "application": "ambassador", "notices": [], "cached": false, "timestamp": 1539606411.061929}
+
+    2018-10-10 12:26:54 diagd 0.40.0 [P15TMainThread] INFO: thread count 3, listening on 0.0.0.0:8877
+    [2018-10-10 12:26:54 +0000] [15] [INFO] Starting gunicorn 19.8.1
+    [2018-10-10 12:26:54 +0000] [15] [INFO] Listening at: http://0.0.0.0:8877 (15)
+    [2018-10-10 12:26:54 +0000] [15] [INFO] Using worker: threads
+    [2018-10-10 12:26:54 +0000] [42] [INFO] Booting worker with pid: 42
+    2018-10-10 12:26:54 diagd 0.40.0 [P42TMainThread] INFO: Starting periodic updates
+    [2018-10-10 12:27:01.977][21][info][main] source/server/drain_manager_impl.cc:63] shutting down parent after drain
+    ```
+
+## Examine Pod and Container Contents
+
+You can examine the contents of the Ambassador Pod for issues, such as if volume mounts are correct and TLS certificates are present in the required directory, to determine if the Pod has the latest Ambassador configuration, or if the generated Envoy configuration is correct or as expected. In these instructions, we will look for problems related to the Envoy configuration.
+
+1. To look into an Ambassador Pod, use the container shell with the `kube-exec` and the `/bin/sh` commands. For example, `kubectl exec -it ambassador-<pod> -- /bin/sh`
```

Comment:
```
kubectl exec -it -n ambassador <pod_name> -- /bin/bash
```

It will not spit out a listing of any kind. It'll give you shell prompt at which you can type commands.